src/rust/engine/src/rule_graph.rs

```diff
@@ -423,11 +427,7 @@ impl <'a> GraphMaker<'a> {
         }
       }
     }
-    RuleGraph {
-      rule_dependency_edges: new_dependency_edges,
-      root_dependencies: new_root_dependencies,
-      unfulfillable_rules: new_unfulfillable_rules
-    }
+    rule_graph
```

Comment:
Hm... you're taking a clone, making it locally mutable, and then returning it. Easier would be to take a mutable reference, and then not return a value?